recipes/full_finetune_distributed.py

```diff
@@ -946,11 +946,13 @@ def recipe_main(cfg: DictConfig) -> None:
             "Distributed finetune recipe should be run via a distributed launcher."
             "If using tune CLI, please specify --nnodes 1 and --nproc_per_node [num_gpus]"
         )
-    init_process_group(backend="gloo" if cfg.device == "cpu" else "nccl")
+    process_group = "gloo" if cfg.device == "cpu" else "nccl"
     if cfg.get("fsdp_cpu_offload", False):
         # Utilize all available CPU cores for intra-op parallelism. This provides ~2x
         # speed up when benchmarking fused AdamW on CPU
         training.set_torch_num_threads()
+        process_group = "cuda:nccl,cpu:gloo"
```

Comment:
n00b question: why not just make this the default every time, instead of `"gloo" if cfg.device == "cpu" else "nccl"`

Reply:
<img width="1413" alt="image" src="https://github.com/user-attachments/assets/be6db7b2-9ffc-4894-b9af-26e39ca94aa7">

Reply:
I actually wasn't sure myself why we do this. Well [now I know](https://github.com/pytorch/torchtune/pull/482#discussion_r1529635443). So yes, I think we can take your suggestion here